buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -0,0 +1,1297 @@
+		// Official benchmark results are meant to be displayed on PerfKitExplorer dashboards.
+		String isOfficial = System.getProperty('official', 'false')
+
+		// Specifies names of benchmarks to be run by PerfKitBenchmarker.
+		String benchmarks = System.getProperty('benchmarks', 'beam_integration_benchmark')
+
+		// If beam is not "prebuilt" then PerfKitBenchmarker runs the build task before running the tests.
+		String beamPrebuilt = System.getProperty('beamPrebuilt', 'true')
+
+		// Beam's sdk to be used by PerfKitBenchmarker.
+		String beamSdk = System.getProperty('beamSdk', 'java')
+
+		// Timeout (in seconds) after which PerfKitBenchmarker will stop executing the benchmark (and will fail).
+		String timeout = System.getProperty('itTimeout', '1200')
+
+		// Path to kubernetes configuration file.
+		String kubeconfig = System.getProperty('kubeconfig', System.getProperty('user.home') + '/.kube/config')
+
+		// Path to kubernetes executable.
+		String kubectl = System.getProperty('kubectl', 'kubectl')
+
+		// Paths to files with kubernetes infrastructure to setup before the test runs.
+		// PerfKitBenchmarker will have trouble reading 'null' path. It expects empty string if no scripts are expected.
+		String kubernetesScripts = System.getProperty('kubernetesScripts', '')
+
+		// Path to file with 'dynamic' and 'static' pipeline options.
+		// that will be appended by PerfKitBenchmarker to the test running command.
+		// PerfKitBenchmarker will have trouble reading 'null' path. It expects empty string if no config file is expected.
+		String optionsConfigFile = System.getProperty('beamITOptions', '')
+
+		// Any additional properties to be appended to benchmark execution command.
+		String extraProperties = System.getProperty('beamExtraProperties', '')
+
+		// Runner which will be used for running the tests. Possible values: dataflow/direct.
+		// PerfKitBenchmarker will have trouble reading 'null' value. It expects empty string if no config file is expected.
+		String runner = System.getProperty('integrationTestRunner', '')
+
+		// Filesystem which will be used for running the tests. Possible values: hdfs.
+		// if not specified runner's local filesystem will be used.
+		String filesystem = System.getProperty('filesystem')
+
+		/* Always required properties: */
+
+		// Pipeline options to be used by the tested pipeline.
+		String integrationTestPipelineOptions = System.getProperty('integrationTestPipelineOptions')
+
+		// Fully qualified name of the test to be run, eg:
+		// 'org.apache.beam.sdks.java.io.jdbc.JdbcIOIT'.
+		String integrationTest = System.getProperty('integrationTest')
+
+		// Relative path to module where the test is, eg. 'sdks/java/io/jdbc.
+		String itModule = System.getProperty('itModule')
+	}
+
+	// A class defining the set of configurable properties accepted by containerImageName.
+	class ContainerImageNameConfiguration {
+		String root = null // Sets the docker repository root (optional).
```

Comment:
nit: null is the default already so we don't need to explicitly set it

Reply:
I like it explicit, but also this is a non-diff from build_rules.gradle.

Reply:
Its just noise since groovy/java developers know what the default values for unassigned variables are.